File: algo/InsertSort/insertsort.go
package main

import "fmt"

func insertsort(arr []int) {
	for index := 1; index < len(arr); index++ {
		visitor := index - 1
		tempVal := arr[index]
		for visitor > -1 && arr[visitor] > tempVal {
			arr[visitor+1] = arr[visitor]
			visitor--
		}
		arr[visitor+1] = tempVal
		fmt.Println(arr)
	}
}
func main() {
	arr := []int{5, 2, 8, 6, 1, 7, 4, 3}
	insertsort(arr)
	fmt.Println("Ans = ", arr)
}

File: algo/InsertSort/insertsort.py
class InsertSort:
    @classmethod
    def sort(cls, arr: list[int]) -> list[int]:
        for index in range(1, len(arr)):
            visitor = index - 1
            cur_val = arr[index]
            while visitor >= 0 and arr[visitor] > cur_val:
                arr[visitor + 1] = arr[visitor]
                visitor -= 1
            arr[visitor + 1] = cur_val
        return arr

if __name__ == "__main__":
    nums = [5, 8, 6, 3, 9, 1]
    print(InsertSort.sort(nums))